cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -490,7 +601,18 @@ func (scaleSet *ScaleSet) DeleteNodes(nodes []*apiv1.Node) error {
 		ref := &azureRef{
 			Name: node.Spec.ProviderID,
 		}
-		refs = append(refs, ref)
+
+		if node.Annotations[cloudprovider.FakeNodeReasonAnnotation] == cloudprovider.FakeNodeUnregistered {
```

Comment:
Not required in master due to 1.30 fork stop cherry-picking the difference (??). This one might worth looking into---later.